Common/Trees/src/main/java/org/tribuo/common/tree/AbstractTrainingNode.java

```diff
@@ -41,33 +43,71 @@
 
     protected double impurityScore;
     
-    protected AbstractTrainingNode<T> greaterThan;
+    protected Node<T> greaterThan;
     
-    protected AbstractTrainingNode<T> lessThanOrEqual;
+    protected Node<T> lessThanOrEqual;
 
-    protected AbstractTrainingNode(int depth, int numExamples) {
+    protected AbstractTrainingNode(int depth, int numExamples, LeafDeterminer leafDeterminer) {
         this.depth = depth;
         this.numExamples = numExamples;
+        this.leafDeterminer = leafDeterminer;
     }
 
     /**
      * Builds next level of a tree.
      * @param featureIDs Indices of the features available in this split.
      * @param rng Splittable random number generator.
      * @param useRandomSplitPoints Whether to choose split points for features at random.
-     * @param scaledMinImpurityDecrease The product of the weight sum of the original examples and the
-     *                                  minImpurityDecrease.
      * @return A possibly empty list of TrainingNodes.
      */
     public abstract List<AbstractTrainingNode<T>> buildTree(int[] featureIDs, SplittableRandom rng,
-                                                            boolean useRandomSplitPoints, float scaledMinImpurityDecrease);
+                                                            boolean useRandomSplitPoints);
 
     public abstract Node<T> convertTree();
 
+    public abstract float getWeightSum();
```

Comment:
Bit of javadoc for the new method please.

Reply:
Done.